consumer.go

```diff
@@ -669,6 +669,11 @@ func (r *Consumer) startStopContinueBackoff(conn *Conn, success bool) {
 			var i int
 			conns := r.conns()
 			if len(conns) == 0 {
+				r.backoffMtx.Lock()
+				if r.backoffCounter > 0 {
+					r.backoffCounter--
+				}
+				r.backoffMtx.Unlock()
```

Comment:
Thanks for identifying this bug.

Unfortunately, I'm not convinced this is the right fix.  I *think* the correct behavior should be for this to loop until it has a connection (also paying attention to an exit signal), and then send `RDY 1`.

Thoughts @jehiah